### DISCUSSION TITLE
group-disabled

### DISCUSSION THREAD

**Discussion by jayenne** [tailwindlabs/tailwindcss#7085] [Ideas]
hiya. one for the wishlins... group-disabled. ty.

**Comment by windyclory** [comment]
What is this for? Can you please explain it a little bit?

**Reply by jayenne** [reply]
sure. for elemnts that support the 'disabled' attribute that can also contain further html such as button, fieldset, optgroup, option and select. It would be nice to have the child elements inherit the disabled attribute for syling purposes.

In this hypothetical, the disabled button has some inner-elements that have their own individual styling. Currently, these style are ignorant to the parent's disabled status and so can't benefit from using it to style itself e.g...'disabled:bg-slate-200' will have no effect...

```
<button disabled class="bg-slate-200 text-blue-500 disabled:bg-slate-200 disabled:text-slate-500">
  <div class="text-white bg-purple-500 disabled:bg-slate-200 disabled:text-slate-500">hello</div>
  <div class="text-white bg-indigo-500">world</div>
</button>
```

whereas, a proposed group-dissabled pseudo-class would allow us to style the inner elements based on the state of the parent element's disabled status...

```
<button disabled class="bg-slate-200 text-blue-500 disabled:bg-slate-200 disabled:text-slate-500">
  <div class="text-white bg-purple-500 group-disabled:bg-slate-200 group-disabled:text-slate-500">hello</div>
  <div class="text-white bg-indigo-500 group-disabled:bg-slate-200 group-disabled:text-slate-500">world</div>
</button>
```
hope it makes sense.


p.s..
[In my [perfectly self-centered user-case (laravel) ]
I have an svg icon component within a button where I need the svg to inherit the different styling of both enabled/disabled cases BUT, can't cleanly pass any variables within the class to facilitate this effect...
```
<button disabled class="bg-slate-200 text-blue-500 disabled:bg-slate-200 disabled:text-slate-500">
  <x-svg-light.user class="w-auto h-5 mr-2 bg-green-500 text-white"/><div class="text-white bg-purple-500 group-disabled:bg-slate-200 group-disabled:text-slate-500">hello world</div>
</button>
```
